backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ClientProvider.java

```diff
@@ -18,9 +18,71 @@
  ****************************************************************/
 package org.apache.james.backends.es;
 
+import java.io.IOException;
+import java.time.Duration;
+import java.time.LocalDateTime;
+
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import org.apache.commons.lang3.time.DurationFormatUtils;
+import org.apache.http.HttpHost;
+import org.elasticsearch.client.RestClient;
 import org.elasticsearch.client.RestHighLevelClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class ClientProvider implements Provider<RestHighLevelClient> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientProvider.class);
+
+    private static final String HTTP_HOST_SCHEME = "http";
+    private final ElasticSearchConfiguration configuration;
+    private final RestHighLevelClient client;
+
+    @Inject
+    @VisibleForTesting
+    ClientProvider(ElasticSearchConfiguration configuration) {
+        this.configuration = configuration;
+        this.client = connect(configuration);
+    }
+
+    private RestHighLevelClient connect(ElasticSearchConfiguration configuration) {
+        Duration waitDelay = Duration.ofMillis(configuration.getMinDelay());
+        Duration forever = Duration.ofMillis(Long.MAX_VALUE);
```

Comment:
FOREVER  could be a constant

Reply:
Already discussed in another PR: I don't see the added value to move that into a broader context. BTW, this code has just been moved and is not really part of this PR.

Reply:
ok, it's not a big deal